src/common/buffer.cc

```diff
@@ -1780,6 +1780,18 @@ static simple_spinlock_t buffer_debug_lock = SIMPLE_SPINLOCK_INITIALIZER;
     return _buffers.front().c_str();  // good, we're already contiguous.
   }
 
+  string buffer::list::to_str() const {
+    string s;
+    for (std::list<ptr>::const_iterator p = _buffers.begin();
+	 p != _buffers.end();
+	 ++p) {
+      if (p->length()) {
+	s += string(p->c_str(), p->length());
```

Comment:
consider using `s.append(p->c_str(), p->length())` to avoid allocating and copying from a temporary string. if you know the total length, you can also use `s.reserve(len)` at the beginning to avoid extra allocations on append